lib/loader/index.d.ts

```diff
@@ -84,6 +84,47 @@ export interface ASUtil {
   __getFloat32ArrayView(ptr: number): Float32Array;
   /** Gets a live view on a Float64Array's values in the module's memory. */
   __getFloat64ArrayView(ptr: number): Float64Array;
+  /** Gets a live view on a StaticArray<i8>'s values in the module's memory. */
+  __getInt8StaticArrayView(ptr: number): Int8Array;
+  /** Gets a live view on a StaticArray<u8>'s values in the module's memory. */
+  __getUint8StaticArrayView(ptr: number): Uint8Array;
+  /** Gets a live view on a StaticArray<i16>'s values in the module's memory. */
+  __getInt16StaticArrayView(ptr: number): Int16Array;
+  /** Gets a live view on a StaticArray<u16>'s values in the module's memory. */
+  __getUint16StaticArrayView(ptr: number): Uint16Array;
+  /** Gets a live view on a StaticArray<i32>'s values in the module's memory. */
+  __getInt32StaticArrayView(ptr: number): Int32Array;
+  /** Gets a live view on a StaticArray<u32>'s values in the module's memory. */
+  __getUint32StaticArrayView(ptr: number): Uint32Array;
+  /** Gets a live view on a StaticArray<i64>'s values in the module's memory. */
+  __getInt64StaticArrayView?(ptr: number): BigInt64Array;
+  /** Gets a live view on a StaticArray<u64>'s values in the module's memory. */
+  __getUint64StaticArrayView?(ptr: number): BigUint64Array;
+  /** Gets a live view on a StaticArray<f32>'s values in the module's memory. */
+  __getFloat32StaticArrayView(ptr: number): Float32Array;
+  /** Gets a live view on a StaticArray<f64>'s values in the module's memory. */
+  __getFloat64StaticArrayView(ptr: number): Float64Array;
+
+    /** Copies an StaticArray<i8>'s values from the module's memory. */
+    __getInt8StaticArray(ptr: number): Int8Array;
+    /** Copies an StaticArray<u8>'s values from the module's memory. */
+    __getUint8StaticArray(ptr: number): Uint8Array;
+    /** Copies an StaticArray<i16>'s values from the module's memory. */
+    __getInt16StaticArray(ptr: number): Int16Array;
+    /** Copies an StaticArray<u16>'s values from the module's memory. */
+    __getUint16StaticArray(ptr: number): Uint16Array;
+    /** Copies an StaticArray<i32>'s values from the module's memory. */
+    __getInt32StaticArray(ptr: number): Int32Array;
+    /** Copies an StaticArray<u32>'s values from the module's memory. */
+    __getUint32StaticArray(ptr: number): Uint32Array;
+    /** Copies an StaticArray<i64>'s values from the module's memory. */
+    __getInt64StaticArray?(ptr: number): BigInt64Array;
+    /** Copies an StaticArray<u64>'s values from the module's memory. */
+    __getUint64StaticArray?(ptr: number): BigUint64Array;
+    /** Copies a StaticArray<f32>'s values from the module's memory. */
+    __getFloat32StaticArray(ptr: number): Float32Array;
+    /** Copies a StaticArray<f64>'s values from the module's memory. */
+    __getFloat64StaticArray(ptr: number): Float64Array;
```

Comment:
Makes me wonder if we should instead expose an enum of types and have a single function to do this, like `__getStaticArray(theType, ptr)`?

Reply:
My suggestion was to modify getTypedArray() to accept static array pointers. Would this suffice?

Reply:
Also we need to handle static arrays of references...